pf4j/src/main/java/org/pf4j/Extension.java

```diff
@@ -34,4 +34,13 @@
 
     int ordinal() default 0;
 
+    /**
+     * An array of extension points, that are implemented by this extension.
+     * This explicit configuration overrides the automatic detection of extension points in the
+     * {@link org.pf4j.processor.ExtensionAnnotationProcessor}.
+     *
+     * @return classes of extension points implemented by this extension
+     */
+    Class<? extends ExtensionPoint>[] provides() default {};
```

Comment:
Here I have two questions:
- do you think that it's better to rename this attribute from `provides` in `points` (I feel that it's better from semantically point of view) ?
- I don't have until now an extension that implements multiple points. I don't know if it's possible with the actual release and I didn't tested. Personal I prefer in my projects to have one extension for one extension point. So, my question is do we real need a list of extension point class for this attribute, is not a single extension point class enough (at least for the moment)?

Reply:
> do you think that it's better to rename this attribute from `provides` in `points` (I feel that it's better from semantically point of view) ?

I'm also fine with `points`. I guess that's even more precise. I've [updated the pull request](https://github.com/pf4j/pf4j/pull/265/commits/33558c798efa0238ebdd30dab018be5570654c8f) accordingly.

> I don't have until now an extension that implements multiple points. Personal I prefer in my projects to have one extension for one extension point. 

It's the same for me. I basically always use the pattern described in issue #264 and don't make use of multiple extension points in a single extension class.

> So, my question is do we real need a list of extension point class for this attribute, is not a single extension point class enough (at least for the moment)?

Even if I don't use this approach, I would suggest to keep the possibility to use multiple extension points in a single extension. 

Java allows multiple inheritance of interfaces. As far as I know this feature used from time to time. If PF4J wouldn't support this approach with its extension, it would restrict plugin developers in their possibilities. As long as there is no definitive technical reason of not supporting this feature, I would suggest to support it with PF4J.

> I don't know if it's possible with the actual release and I didn't tested.

I also didn't test it. But as far as I've seen / understood the source code of PF4J, I guess it should / could work. Why don't we just add this scenario to the demo application? - Afterwards we should no better. ;)